### ISSUE TITLE
Convert packer templates to HCL

### ISSUE THREAD

**Issue by brucellino** [EGI-Federation/fedcloud-vmi-templates#72]
# Short Description of the issue

Hello! I see lots of JSON templates. In order to familiarise myself with the repo, I'd like to propose that I contribute the HCL version of the Packer templates.

# Summary of proposed changes

Convert all the templates in centos/ and ubuntu/ to pkr.hcl format.

**Comment by sebastian-luna-valero** [comment]
Currently looking at adding Alma 9 (xref: https://github.com/EGI-Federation/fedcloud-vmi-templates/issues/68#issuecomment-2273435591)

Before I read about it, @brucellino could you please summarize what's the advantage of moving the packer templates from `json` to `pkr.hcl` format?

**Comment by brucellino** [comment]
Short version is that JSON is [legacy](https://developer.hashicorp.com/packer/docs/templates/legacy_json_templates) and won't be extended or guaranteed to work.

Long version is that you get HCL (_i.e._ an actual language for configuration)

- you can use the [HCL functions](https://developer.hashicorp.com/packer/docs/templates/hcl_templates/functions) (Read secrets, convert data, lookup data sources, _etc_). 
  - You can factorise the templates... especially helpful for longer provisioners.
- you can use [HCL Variables with validation](https://developer.hashicorp.com/packer/docs/templates/hcl_templates/variables) (you write an actual template instead of a single statement which has to be modified per deploy).
- I mean... Comments in the file? that's pretty cool right.
- you can still convert to JSON on the fly if that's your thing.

So, basically you lose nothing, but you get cleaner and safer specification.

**Comment by sebastian-luna-valero** [comment]
Done in https://github.com/EGI-Federation/fedcloud-vmi-templates/pull/87
